daphne/src/error/mod.rs

```diff
@@ -65,9 +65,10 @@ impl FatalDapError {
 impl From<VdafError> for DapError {
     fn from(e: VdafError) -> Self {
         match e {
-            VdafError::Codec(..) | VdafError::Vdaf(..) | VdafError::Uncategorized(..) => {
+            VdafError::Codec(..) | VdafError::Vdaf(..) => {
                 Self::Transition(TransitionFailure::VdafPrepError)
             }
+            VdafError::Uncategorized(s) => Self::Fatal(FatalDapError(s)),
         }
     }
 }
```

Comment:
Realllly great suggestion! I think I know how I want to address this, but the implementation will take some time. I'll tack it on as a commit and  ping you when it's ready.

Reply:
I've slowly come to the realizations that having `Unknown` or `Other` variants on enums is very rarely a good idea. I also used to do it. But it creates situations like this, where infalible functions become falible because you have to handle a case that almost never or never happens and should have been handled earlier anyway

Reply:
I've replaced `Uncategorized(String)` with `Dap(DapError)` and cleaned things up accordingly.